src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverter.FromYCbCrVector4.cs

```diff
@@ -20,73 +20,69 @@ public FromYCbCrVector4(int precision)
 
             protected override bool IsAvailable => SimdUtils.HasVector4;
 
-            protected override void ConvertCoreVectorized(in ComponentValues values, Span<Vector4> result)
+            protected override void ConvertCoreVectorizedInplace(in ComponentValues values)
             {
-                // TODO: Find a way to properly run & test this path on AVX2 PC-s! (Have I already mentioned that Vector<T> is terrible?)
-                DebugGuard.IsTrue(result.Length % 8 == 0, nameof(result), "result.Length should be divisible by 8!");
+                DebugGuard.IsTrue(values.Component0.Length % 8 == 0, nameof(values), "Length should be divisible by 8!");
 
-                ref Vector4Pair yBase =
+                ref Vector4Pair c0Base =
                     ref Unsafe.As<float, Vector4Pair>(ref MemoryMarshal.GetReference(values.Component0));
-                ref Vector4Pair cbBase =
+                ref Vector4Pair c1Base =
                     ref Unsafe.As<float, Vector4Pair>(ref MemoryMarshal.GetReference(values.Component1));
-                ref Vector4Pair crBase =
+                ref Vector4Pair c2Base =
                     ref Unsafe.As<float, Vector4Pair>(ref MemoryMarshal.GetReference(values.Component2));
 
-                ref Vector4Octet resultBase =
-                    ref Unsafe.As<Vector4, Vector4Octet>(ref MemoryMarshal.GetReference(result));
-
                 var chromaOffset = new Vector4(-this.HalfValue);
                 var maxValue = this.MaximumValue;
 
                 // Walking 8 elements at one step:
-                int n = result.Length / 8;
+                int n = values.Component0.Length / 8;
```

Comment:
```suggestion
                int n = (int)((uint)values.Component0.Length / 8);
```

Reply:
I'm not sure it's worth it, see https://github.com/SixLabors/ImageSharp/pull/1773#discussion_r720905777